### PR TITLE
reorganize and add dockerfile

### PR DESCRIPTION
**Pull request by bmw2621**
Added Dockerfile. Built image tested.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1 @@
+Dockerfile
```
Comment (TheSavageDev) [inline .dockerignore]: Add node_modules, build files, and any log files

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.env
+/node_modules
```
Comment (TheSavageDev) [inline .gitignore:2]: Add logs and build files

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "nodemon": "^2.0.6"
   },
   "scripts": {
-    "start": "nodemon app.js"
+    "start": "nodemon app.js",
```
Comment (TheSavageDev) [inline package.json]: This won't work anymore after moving app.is

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FROM node:lts-alpine3.12
+
+ADD ./src .
+
+ENTRYPOINT [ "node", "src/app.js" ]
```
Comment (TheSavageDev) [inline Dockerfile]: You should create a working directory to put your code in. 

`WORKDIR /use/src/app`

Then copy over package.json and package-lock.json

`COPY package*.json ./`

Then install dependencies

`RUN npm install`

If you're running in prod you should do `npm ci` but we can talk about why later. 

Then copy over source code

`COPY . .`

Expose your port

`EXPOSE 5000`

Finally your start command

`CMD ["node", "src/app.is"]`

**Review by TheSavageDev — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (bmw2621) [reply Dockerfile]: Done

File: .dockerignore (hunk 1 above)
Comment (bmw2621) [reply .dockerignore]: No log or build files are produced.  Need to add logging to the app.

File: .gitignore (hunk 2 above)
Comment (bmw2621) [reply .gitignore:2]: No log or build files are produced. Need to add logging to the app.

**Review by TheSavageDev — APPROVED** [review]

**Review by TheSavageDev — APPROVED** [review]

**Pull request merged**